contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -0,0 +1,1122 @@
+    int index = tupleWriter.tupleSchema().index(fieldName);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, type, mode);
+      index = tupleWriter.addColumn(colSchema);
+    }
+    return tupleWriter.scalar(index);
+  }
+
+  /**
+   * This helper function gets the attributes for an HDF5 datapath.  These attributes are projected as a map in select * queries when the defaultPath is null.
+   * @param rowWriter the row to which the data will be written
+   * @param record the record for the attributes
+   */
+  private void writeAttributes(TupleWriter rowWriter, HDF5DrillMetadata record) {
+    Map attribs = getAttributes(record.getPath());
+    Iterator entries = attribs.entrySet().iterator();
+
+    int index = rowWriter.tupleSchema().index("attributes");
+    if (index == -1) {
+      index = rowWriter
+        .addColumn(SchemaBuilder.columnSchema("attributes", TypeProtos.MinorType.MAP, TypeProtos.DataMode.REQUIRED));
+    }
+    TupleWriter mapWriter = rowWriter.tuple(index);
+
+    while (entries.hasNext()) {
+      Map.Entry entry = (Map.Entry) entries.next();
+      String key = (String) entry.getKey();
+
+      HDF5Attribute attrib = (HDF5Attribute) entry.getValue();
+      switch (attrib.getDataType()) {
+        case BIT:
+          boolean value = (Boolean) attrib.getValue();
+          writeBooleanColumn(mapWriter, key, value);
+          break;
+        case BIGINT:
+          writeLongColumn(mapWriter, key, (Long) attrib.getValue());
+          break;
+        case INT:
+          writeIntColumn(mapWriter, key, (Integer) attrib.getValue());
+          break;
+        case FLOAT8:
+          writeFloat8Column(mapWriter, key, (Double) attrib.getValue());
+          break;
+        case FLOAT4:
+          writeFloat4Column(mapWriter, key, (Float) attrib.getValue());
+          break;
+        case VARCHAR:
+          writeStringColumn(mapWriter, key, (String) attrib.getValue());
+          break;
+        case TIMESTAMP:
+          writeTimestampColumn(mapWriter, key, (Long) attrib.getValue());
+        case GENERIC_OBJECT:
+          //This is the case for HDF5 enums
+          String enumText = attrib.getValue().toString();
+          writeStringColumn(mapWriter, key, enumText);
+          break;
+      }
+    }
+  }
+
+  /**
+   * This function processes the MAP data type which can be found in HDF5 files.
+   * It automatically flattens anything greater than 2 dimensions.
+   *
+   * @param path the HDF5 path tp the compound data
+   * @param fieldNames The field names to be mapped
+   * @param reader the HDF5 reader for the data file
+   * @param rowWriter the rowWriter to write the data
+   */
+
+  private void getAndMapCompoundData(String path, List<String> fieldNames, IHDF5Reader reader, RowSetLoader rowWriter) {
+
+    final String COMPOUND_DATA_FIELD_NAME = "compound_data";
+    path = HDF5Utils.resolvePath(path, reader);
```

Comment:
I am wondering why this method set class variable inside.
It should either return it and assign when method is called or not assign at all.
Anyway, step with the assignment should not be hidden.

Reply:
Yeah... fixed that.